slsim/Sources/random_supernovae.py

```diff
@@ -68,26 +93,36 @@ def __init__(
                 + ", ".join(accepted_types)
             )
         self._sn_type = sn_type
-        self._accepted_SN_types = accepted_types
-        self._all_sncosmo_models = all_models
-        self._type_models = None
         self._absolute_mag_band = absolute_mag_band
 
-        self.set_random_sed_model(self._sn_type)
-
         if absolute_mag is None:
             absolute_mag = self.get_absolute_magnitude(self._sn_type)
 
-        super(RandomizedSupernova, self).__init__(
+        if modeldir is None:
+            self._accepted_SN_types = accepted_types
+            self._all_sncosmo_models = all_models
+            self._type_models = None
+            self.set_random_sed_model(self._sn_type)
+        elif sn_type == "Ia":
+            self._sncosmo_source = "salt3"
+        else:
+            self._sncosmo_source = str(
+                random.choice(os.listdir(os.path.join(modeldir, sn_type)))
```

Comment:
would there be a numpy equivalent to random.choice() as there we could initialize a random seed, if needed